x-pack/plugins/observability/public/hooks/use_fetch_rules.ts

```diff
@@ -33,6 +34,23 @@ export function useFetchRules({
 
   const [noData, setNoData] = useState<boolean>(true);
   const [initialLoad, setInitialLoad] = useState<boolean>(true);
+  const [tagsState, setTagsState] = useState<TagsState>({
+    data: [],
+    error: null,
+  });
+  const loadRuleTagsAggs = useCallback(async () => {
+    try {
+      const ruleTagsAggs = await loadRuleTags({
+        http,
+      });
+
+      if (ruleTagsAggs?.ruleTags) {
+        setTagsState({ data: ruleTagsAggs.ruleTags, error: null });
+      }
+    } catch (e) {
+      setTagsState((oldState) => ({ ...oldState, error: RULE_TAGS_LOAD_ERROR }));
```

Comment:
Use the type you already defined, `TagsState` for `oldState`.
```suggestion
      setTagsState((oldState: TagsState) => ({ ...oldState, error: RULE_TAGS_LOAD_ERROR }));
```